File: java_sdk/src/main/java/business_api_client/OpenApiv13adgroupupdateAudienceRule.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import business_api_client.OpenApiv13adgroupcreateAudienceRuleInclusions;
import business_api_client.OpenApiv13adgroupupdateAudienceRuleExclusions;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
/**
 * List of audience IDs. You can get audience IDs by using the [/dmp/custom_audience/list/](https://ads.tiktok.com/marketing_api/docs?id&#x3D;1739940506015746) endpoint.
 */
@Schema(description = "List of audience IDs. You can get audience IDs by using the [/dmp/custom_audience/list/](https://ads.tiktok.com/marketing_api/docs?id=1739940506015746) endpoint.")
@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2023-06-28T14:49:22.099759+05:30[Asia/Kolkata]")
public class OpenApiv13adgroupupdateAudienceRule {
  @JsonProperty("exclusions")
  private OpenApiv13adgroupupdateAudienceRuleExclusions exclusions = null;

  @JsonProperty("inclusions")
  private OpenApiv13adgroupcreateAudienceRuleInclusions inclusions = null;

  public OpenApiv13adgroupupdateAudienceRule exclusions(OpenApiv13adgroupupdateAudienceRuleExclusions exclusions) {
    this.exclusions = exclusions;
    return this;
  }

   /**
   * Get exclusions
   * @return exclusions
  **/
  @Schema(description = "")
  public OpenApiv13adgroupupdateAudienceRuleExclusions getExclusions() {
    return exclusions;
  }

  public void setExclusions(OpenApiv13adgroupupdateAudienceRuleExclusions exclusions) {
    this.exclusions = exclusions;
  }

  public OpenApiv13adgroupupdateAudienceRule inclusions(OpenApiv13adgroupcreateAudienceRuleInclusions inclusions) {
    this.inclusions = inclusions;
    return this;
  }

   /**
   * Get inclusions
   * @return inclusions
  **/
  @Schema(description = "")
  public OpenApiv13adgroupcreateAudienceRuleInclusions getInclusions() {
    return inclusions;
  }

  public void setInclusions(OpenApiv13adgroupcreateAudienceRuleInclusions inclusions) {
    this.inclusions = inclusions;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    OpenApiv13adgroupupdateAudienceRule openApiv13adgroupupdateAudienceRule = (OpenApiv13adgroupupdateAudienceRule) o;
    return Objects.equals(this.exclusions, openApiv13adgroupupdateAudienceRule.exclusions) &&
        Objects.equals(this.inclusions, openApiv13adgroupupdateAudienceRule.inclusions);
  }

  @Override
  public int hashCode() {
    return Objects.hash(exclusions, inclusions);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class OpenApiv13adgroupupdateAudienceRule {\n");
    
    sb.append("    exclusions: ").append(toIndentedString(exclusions)).append("\n");
    sb.append("    inclusions: ").append(toIndentedString(inclusions)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: js_sdk/src/model/OpenApiv13adgroupupdateAudienceRule.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';
import {OpenApiv13adgroupcreateAudienceRuleInclusions} from './OpenApiv13adgroupcreateAudienceRuleInclusions.js';
import {OpenApiv13adgroupupdateAudienceRuleExclusions} from './OpenApiv13adgroupupdateAudienceRuleExclusions.js';

/**
 * The OpenApiv13adgroupupdateAudienceRule model module.
 * @module model/OpenApiv13adgroupupdateAudienceRule
 * @version 0.1.2
 */
export class OpenApiv13adgroupupdateAudienceRule {
  /**
   * Constructs a new <code>OpenApiv13adgroupupdateAudienceRule</code>.
   * List of audience IDs. You can get audience IDs by using the [/dmp/custom_audience/list/](https://ads.tiktok.com/marketing_api/docs?id&#x3D;1739940506015746) endpoint.
   * @alias module:model/OpenApiv13adgroupupdateAudienceRule
   * @class
   */
  constructor() {
  }

  /**
   * Constructs a <code>OpenApiv13adgroupupdateAudienceRule</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/OpenApiv13adgroupupdateAudienceRule} obj Optional instance to populate.
   * @return {module:model/OpenApiv13adgroupupdateAudienceRule} The populated <code>OpenApiv13adgroupupdateAudienceRule</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new OpenApiv13adgroupupdateAudienceRule();
      if (data.hasOwnProperty('exclusions'))
        obj.exclusions = OpenApiv13adgroupupdateAudienceRuleExclusions.constructFromObject(data['exclusions']);
      if (data.hasOwnProperty('inclusions'))
        obj.inclusions = OpenApiv13adgroupcreateAudienceRuleInclusions.constructFromObject(data['inclusions']);
    }
    return obj;
  }
}

/**
 * @member {module:model/OpenApiv13adgroupupdateAudienceRuleExclusions} exclusions
 */
OpenApiv13adgroupupdateAudienceRule.prototype.exclusions = undefined;

/**
 * @member {module:model/OpenApiv13adgroupcreateAudienceRuleInclusions} inclusions
 */
OpenApiv13adgroupupdateAudienceRule.prototype.inclusions = undefined;

